Kernel/Process.cpp

```diff
@@ -273,6 +273,8 @@ ErrorOr<void> Process::attach_resources(NonnullOwnPtr<Memory::AddressSpace>&& pr
 
     auto weak_ptr = TRY(this->try_make_weak_ptr());
     m_procfs_traits = TRY(ProcessProcFSTraits::try_create({}, move(weak_ptr)));
+    if (fork_parent)
+        m_signal_action_data = fork_parent->m_signal_action_data;
```

Comment:
Hmm...this is a bit weird, does exec() clear these? 
What about pending signals? Should we copy them over?

Reply:
[POSIX mandates](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fork.html) that "The set of signals pending for the child process shall be initialized to the empty set". There seems to be no information about what happens to the signal handlers, so I simply went by existing real-world behaviour for those.

As for whether `exec()` clears those, it doesn't seem like it at the moment. I'll look into that tomorrow.

Reply:
> The set of signals pending for the child process shall be initialized to the empty set

:+1:

Reply:
Shell didn't appear to be broken either way, but I found some POSIX documentation on how to handle signal handlers and implemented that instead.